Zend/zend_call_stack.c

```diff
@@ -664,43 +665,17 @@ static bool zend_call_stack_get_netbsd(zend_call_stack *stack)
 #endif /* defined(__NetBSD__) */
 
 #if defined(__sun)
-# if defined(HAVE_PTHREAD_ATTR_GET_NP) && defined(HAVE_PTHREAD_ATTR_GETSTACK)
 static bool zend_call_stack_get_solaris_pthread(zend_call_stack *stack)
 {
-	pthread_attr_t attr;
-	int error;
-	void *addr;
-	size_t max_size, guard_size;
-
-	error = pthread_attr_get_np(pthread_self(), &attr);
-	if (error) {
-		return false;
-	}
-
-	error = pthread_attr_getstack(&attr, &addr, &max_size);
-	if (error) {
-		return false;
-	}
-
-	error = pthread_attr_getguardsize(&attr, &guard_size);
-	if (error) {
+	stack_t s;
+	if (thr_stksegment(&s) < 0) {
 		return false;
 	}
 
-	addr = (char *)addr + guard_size;
-	max_size -= guard_size;
-
-	stack->base = (char *)addr + max_size;
-	stack->max_size = max_size;
-
+	stack->max_size = s.ss_size;
+	stack->base = (char *)s.ss_sp - stack->max_size;
```

Comment:
`stack->base` should be the high address of the stack (see the diagram in `zend_call_stack_get_linux_proc_maps`), and `ss_sp` is the high address already according to [the manpage](https://man.omnios.org/man3c/thr_stksegment.3c) so I think we shouldn't subtract `stack->max_size` from it here.

Note that in tests, `zend_call_stack_get_solaris` will use `zend_call_stack_get_solaris_proc_maps` and not `zend_call_stack_get_solaris_pthread` because we are the main thread.

Reply:
> `stack->base` should be the high address of the stack 

ah true I forgot.